stdlib/3/multiprocessing/sharedctypes.pyi

```diff
@@ -0,0 +1,133 @@
+def Value(
+    typecode_or_type: Type[ctypes._SimpleCData[_U]], *args: Any, ctx: Optional[context.BaseContext] = ...
+) -> Synchronized[_U]: ...
+@overload
+def Value(
+    typecode_or_type: Type[ctypes.Array[ctypes.c_char]], *args: Any, ctx: Optional[context.BaseContext] = ...
+) -> SynchronizedString: ...
+@overload
+def Value(
+    typecode_or_type: Type[ctypes.Array[_T]], *args: Any, ctx: Optional[context.BaseContext] = ...
+) -> SynchronizedArray[_T]: ...
+
+# Possibly locked case (don't know whether it's synchronized)
+@overload
+def Value(
+    typecode_or_type: str, *args: Any, lock: bool, ctx: Optional[context.BaseContext] = ...
+) -> Union[ctypes._SimpleCData, Synchronized]: ...
+@overload
+def Value(
+    typecode_or_type: Type[ctypes._SimpleCData[_U]], *args: Any, lock: bool, ctx: Optional[context.BaseContext] = ...
+) -> Union[ctypes._SimpleCData[_U], Synchronized[_U]]: ...
+@overload
+def Value(
+    typecode_or_type: Type[ctypes.Array[ctypes.c_char]], *args: Any, lock: bool, ctx: Optional[context.BaseContext] = ...
+) -> Union[ctypes.Array[ctypes.c_char], SynchronizedString]: ...
+@overload
+def Value(
+    typecode_or_type: Type[ctypes.Array[_T]], *args: Any, lock: bool, ctx: Optional[context.BaseContext] = ...
+) -> Union[ctypes.Array[_T], SynchronizedArray[_T]]: ...
+
+# Dynamically generates class, just type as Any
+@overload
+def Value(typecode_or_type: _T, *args: Any, lock: bool = ..., ctx: Optional[context.BaseContext]) -> Any: ...
+
+# Definitely lock case (know it's synchronized)
+@overload
+def Array(
+    typecode_or_type: str, size_or_initializer: Union[int, Sequence[Any]], *, ctx=Optional[context.BaseContext]
+) -> SynchronizedArray[ctypes._SimpleCData]: ...
+@overload
+def Array(
+    typecode_or_type: Type[ctypes.c_char], size_or_initializer: Union[int, bytes], *, ctx=Optional[context.BaseContext]
+) -> SynchronizedString: ...
+@overload
+def Array(
+    typecode_or_type: Type[_T], size_or_initializer: Union[int, Sequence[Any]], *, ctx=Optional[context.BaseContext]
+) -> SynchronizedArray[_T]: ...
+
+# Maybe locked case
+@overload
+def Array(
+    typecode_or_type: str, size_or_initializer: Union[int, Sequence[Any]], *, lock: bool, ctx=Optional[context.BaseContext]
+) -> Union[ctypes.Array[ctypes._SimpleCData], SynchronizedArray[ctypes._SimpleCData]]: ...
+@overload
+def Array(
+    typecode_or_type: Type[ctypes.c_char],
+    size_or_initializer: Union[int, bytes],
+    *,
+    lock: bool,
+    ctx=Optional[context.BaseContext]
+) -> Union[ctypes.Array[ctypes.c_char], SynchronizedString]: ...
+@overload
+def Array(
+    typecode_or_type: Type[_T], size_or_initializer: Union[int, Sequence[Any]], *, lock: bool, ctx=Optional[context.BaseContext]
+) -> Union[ctypes.Array[_T], SynchronizedArray[_T]]: ...
+
+class SynchronizedBase(ContextManager[bool]):
+    def get_lock(self) -> _LockLike: ...
+    def __repr__(self) -> str: ...
+
+class Synchronized(Generic[_U], SynchronizedBase):
+    value: _U
+    def __init__(
```

Comment:
Why aren't `__init__` and `get_obj` on `SynchronizedBase`, like in the implementation?

Reply:
Originally because I couldn't figure out how to get access the right type variable for `value`, but I think I've figured out how to now.